src/main/java/com/mabl/integration/jenkins/MablStepBuilder.java

```diff
@@ -95,55 +112,47 @@ public boolean perform(
                 applicationId,
                 continueOnPlanFailure,
                 continueOnMablError,
-                getOutputFileLocation(build)
+                getOutputFileLocation(workspace)
         );
 
         ExecutorService executorService = Executors.newSingleThreadExecutor();
         Future<Boolean> runnerFuture = executorService.submit(runner);
         try {
-            return runnerFuture.get(EXECUTION_TIMEOUT_SECONDS, SECONDS);
-
+            if(runnerFuture.get(EXECUTION_TIMEOUT_SECONDS, SECONDS)) {
+                run.setResult(Result.SUCCESS);
+            } else {
+                run.setResult(Result.FAILURE);
+            }
         } catch (ExecutionException e) {
             outputStream.println("There was an execution error trying to run your journeys in mabl");
             e.printStackTrace(outputStream);
-            return continueOnMablError;
-
+            run.setResult(Result.FAILURE);
         } catch (TimeoutException e) {
             outputStream.printf("Oh dear. Your journeys exceeded the max plugin runtime limit of %d seconds.%n" +
                     "We've aborted this Jenkins step, but your journeys may still be running in mabl.", EXECUTION_TIMEOUT_SECONDS);
-            return continueOnMablError;
+            run.setResult(Result.FAILURE);
```

Comment:
Are we removing the ability to continue on certain types of errors?

Reply:
no, this was a mistake. Will fix